app/sequencescape_excel/sequencescape_excel/specialised_field/chromium_tag_group.rb

```diff
@@ -4,18 +4,33 @@ module SequencescapeExcel
   module SpecialisedField
     ##
     # ChromiumTagGroup
+    #
+    # This class represents a single index tag group for Chromium.
+    # It includes common functionality from ChromiumTagGroupCommon.
     class ChromiumTagGroup
       include Base
 
       validate :check_tag_group
 
+      ##
+      # Retrieves the ID of the tag group.
+      #
+      # This method finds the tag group ID by searching within the
+      # visible_single_index_chromium scope of the TagSet model.
+      #
+      # @return [Integer, nil] The ID of the tag group, or nil if not found.
       def tag_group_id
-        @tag_group_id ||= ::TagGroup.chromium.find_by(name: value)&.id
+        tag_groups = ::TagSet.tag_groups_within_visible_single_index_chromium
+        @tag_group_id ||= tag_groups.find_by(name: value)&.id
```

Comment:
I think:
* The dropdown in the spreadsheet cell will be a list of Tag Set names
* This code treats the value selected in the spreadsheet as a Tag Group name
* Therefore, if you have a Tag Set linked to a Tag Group that have different names, this won't find it.

Is that right? I think the names are the same on the data you've created so far, but there's no guarantee it will always be the case? Thanks.

Reply:
Good spot 👍